kyaml/resid/gvk.go

```diff
@@ -18,7 +18,7 @@ type Gvk struct {
 	Kind    string `json:"kind,omitempty" yaml:"kind,omitempty"`
 	// isClusterScoped is true if the object is known, per the openapi
 	// data in use, to be cluster scoped, and false otherwise.
-	isClusterScoped bool
+	isClusterScoped bool `json:"-"`
```

Comment:
Could we remove this field altogether and calculate its value on the fly in `IsClusterScoped()` function instead?

When embedding this field, I have run today into one more issue which basically boils down to `unstructured.Unstructured` not being able to handle this object as field is not exported.